components/change_url_modal/change_url_modal.tsx

```diff
@@ -12,73 +11,82 @@ import Constants from 'utils/constants';
 import {getShortenedURL, cleanUpUrlable} from 'utils/url';
 import {t} from 'utils/i18n';
 
-export default class ChangeURLModal extends React.PureComponent {
-    static propTypes = {
+export type Props = {
 
-        /**
-        * Set whether to show the modal or not
-        */
-        show: PropTypes.bool.isRequired,
+    /**
+     * Set whether to show the modal or not
+     */
+    show: boolean;
 
-        /**
-        * Set to change the title of the modal
-        */
-        title: PropTypes.node,
+    /**
+     * Set to change the title of the modal
+     */
+    title: JSX.Element | string;
 
-        /**
-        * Set to change the submit button text
-        */
-        submitButtonText: PropTypes.node,
+    /**
+     * Set to change the submit button text
+     */
+    submitButtonText: JSX.Element | string;
 
-        /**
-        * Set to change the current URL
-        */
-        currentURL: PropTypes.string,
+    /**
+     * Set to change the current URL
+     */
+    currentURL: string;
 
-        /**
-        * Set to the current team URL
-        */
-        currentTeamURL: PropTypes.string.isRequired,
+    /**
+     * Set to the current team URL
+     */
+    currentTeamURL: string;
 
-        /**
-        * Server error from failed channel creation
-        */
-        serverError: PropTypes.node,
+    /**
+     * Server error from failed channel creation
+     */
+    serverError: JSX.Element | string | null;
```

Comment:
Will this work if it's a JSX.Node? That should allow null assignment (not sure about string though).

Reply:
The equivalent for `PropTypes.node` in typescript is actually `React.ReactNode`. Could we change this to the following:
```suggestion
    serverError?: React.ReactNode;
```

Reply:
Fixed !